binary_search_tree/binary_search_tree.py

```diff
@@ -1,14 +1,33 @@
 class BinarySearchTree:
```

Comment:
The Binary Search Tree is actually a very common whiteboard interview question so it would be wise to get it ingrained in your mind.